### ISSUE TITLE
OBJ import missing textures

### ISSUE THREAD

**Issue by zzador** [godotengine/godot#36173]
I have made a little One-Room-Test-Map in GZDoomBuilder with 3 different textures and exported the map to an OBJ file. When importing the OBJ file into Godot it loads 2 of the 3 textures and assigns them to the corresponding walls/floors/ceilings but the third texture (WALLS.PNG) is not loaded. The walls that use this texture stay grey and Godot-Output says:

 `Couldn't open MTL file 'MAP01_MAP01.mtl', it may not exist or not be readable`.

 `No loader found for resource: res://MAP01/WALL1.PNG`

The MTL-File exists and the two other textures are successfully loaded from there, so the first error message makes absolutely no sense.
The texture WALL1.PNG is loaded correctly afterwards as I can edit, see and even manually assign the texture. The textures location is the same as the other two textures and it gets references the same way in the "mtl" file as the other two textures.

I have uploaded the generated OBJ-File and Textures to file-upload:

https://www.file-upload.net/download-13901101/MAP01.zip.html

**Comment by Calinou** [comment]
Can you reproduce this with OBJ files created with another program (such as Blender)?

**Comment by zzador** [comment]
Yes. Now I made two obj files in Wings3D. Both have a cube with 3 different materials. The first file has a mesh-group for every material and the second file has only one mesh-group with 3 material assigns. Both files won't import correctly. It is even worse than the obj file from GZDoomBuilder. Now NO texture is loaded and assigned at all. The same error messages as before.

Files can be downloaded here:
https://www.file-upload.net/download-13901411/meshes.zip.html

**Comment by zzador** [comment]
Bug was found. In this case I would say it is 50/50 case of user-error/software-bug. In the OBJ-File the material library was referenced by "mtllib MAP01/MAP01_MAP01.mtl" but the obj file was already within this directory so the correct reference was "mtllib MAP01_MAP01.mtl". By correcting this Godot was now able to correctly assign every texture. A littlebit surprising and misleading however that Godot managed to assign some of the textures with the wrong reference in the OBJ-File. So it works now but the behaviour with wrong data is a bit confusing. Don't know if I should close this issue.

**Comment by zzador** [comment]
Made a few more tests. Sometimes Godot will import the OBJ files without an error, and the materials are correctly applied and sometimes they are missing and the error messages make absolutely no sense at all. The OBJ-Import seems still very buggy which is really surprising as the OBJ Format is the simplest of all 3D-content formats.

**Comment by Calinou** [comment]
@zzador Can you post the error messages here?

**Comment by git2013vb** [comment]
I have same problem. It is related to godot import I guess:
Godot wireframe:
![image](https://user-images.githubusercontent.com/13235119/77236282-1099f800-6bb5-11ea-9564-2613ccc23fd3.png)

Same obj imported in blender  (wireframe):
![image](https://user-images.githubusercontent.com/13235119/77236307-33c4a780-6bb5-11ea-9fa4-2bc810dcfaff.png)

To many differences if I can say..
Also in Godot there are additional wirefame connection that aren't in obj.

**Comment by zzador** [comment]
> I have same problem. It is related to godot import I guess:
> Godot wireframe:
> ![image](https://user-images.githubusercontent.com/13235119/77236282-1099f800-6bb5-11ea-9564-2613ccc23fd3.png)
> rendered:
> ![image](https://user-images.githubusercontent.com/13235119/77236428-4095cb00-6bb6-11ea-9b74-b401a3b65ae8.png)
> 
> Same obj imported in blender (wireframe):
> ![image](https://user-images.githubusercontent.com/13235119/77236307-33c4a780-6bb5-11ea-9fa4-2bc810dcfaff.png)
> 
> rendered:
> ![image](https://user-images.githubusercontent.com/13235119/77236419-2d82fb00-6bb6-11ea-878f-cac20d23a666.png)
> 
> To many differences if I can say..
> Also in Godot there are additional wirefame connection that aren't in obj.
> 
> Also error for each mtl file. (false errors?)
> 
> ```
> Couldn't open MTL file 'roadTile_001.mtl', it may not exist or not be readable.
> editor/import/resource_importer_obj.cpp:66 - OBJ: Ambient light for material 'Water' is ignored in PB
> ```

Without knowing details I would hardly guess the difference in wireframe is completely normal and the result of triangularization. Godot needs triangles only but Blender works with arbitrary polygons that can have any number of vertices. The Material error however seems to be the same.

**Comment by Calinou** [comment]
@zzador To alleviate issues related to automatic triangulation, it's recommended to use the Triangulate modifier and check Apply Modifiers in Blender's export dialog. This way, you can preview how triangulation will look directly from Blender.

**Comment by zzador** [comment]
> @zzador Can you post the full error log here?
> 
> Also, it would be interesting to try loading the OBJ file at run-time using [gd-obj](https://github.com/Ezcha/gd-obj).

Godot's Output:

 ERROR: No loader found for resource: res://MAP01/WALL1.PNG.
   At: core/io/resource_loader.cpp:285
ERROR: No loader found for resource: res://MAP01/WALL2.JPG.
   At: core/io/resource_loader.cpp:285
ERROR: Couldn't open MTL file 'test.mtl', it may not exist or not be readable.
   At: editor/import/resource_importer_obj.cpp:48
WARNING: _parse_material_library: OBJ: Ambient light for material 'material01' is ignored in PBR
     At: editor/import/resource_importer_obj.cpp:66
WARNING: _parse_material_library: OBJ: Ambient light for material 'material02' is ignored in PBR
     At: editor/import/resource_importer_obj.cpp:66

**Comment by MHillier98** [comment]
Found the same error - I'm not applying any materials in blender, just exporting some text that's been extruded.
Here's what shows when I import:

```
Godot Engine v3.2.2.stable.official
OpenGL ES 3.0 Renderer: GeForce RTX 2080/PCIe/SSE2
 
 Couldn't open MTL file 'Font-7.mtl', it may not exist or not be readable.
 editor/import/resource_importer_obj.cpp:66 - OBJ: Ambient light for material 'None' is ignored in PBR
```

The file is there:
![image](https://user-images.githubusercontent.com/26564949/88801603-44cd9700-d1ed-11ea-8873-f74c6f581ef5.png)

**Comment by aario** [comment]
Any hope this gets fixed?
I also have this issue.
3.4.2 stable
FBX import does nothing.
Obj import gives me a white object with no skin

**Comment by Calinou** [comment]
@aario Please upload a minimal reproduction project so we can make sure this is actually the same issue.

**Comment by aario** [comment]
I would like to. But the problem is I'm quite new to this 3d technology altogether (Although having years of experience in web development). So I have bought and downloaded this:
https://syntystore.com/products/polygon-city-pack
And tried to import it to godot 3.4.2 and godot 4 alpha 4
Since the assets are proprietary I cannot upload them somewhere.

**Comment by aario** [comment]
My problem was with the source file. I used https://github.com/Plattar/gltf-exporter to export Unity assets from inside unity to gltf and now gltf is correctly imported. Althugh those fbx assets which I could not import to godot propely, I wasn't able to import into unity as well.

**Comment by and-rad** [comment]
Regarding the original bug, this seems to be happening because of the order in which the assets are imported. `WALL1` appears after `MAP01` alphabetically, which means that the texture hasn't been imported yet when the material file is being imported. I successfully tested this by renaming various files:

`WALL1.png` -> `LALL01.png`: Import works
`WALL1.png` -> `NALL1.png`: Import doesn't work
`FLOOR1.png` -> `NLOOR1.png`: Import doesn't work

Which would make this issue a duplicate of #28203.
